### DISCUSSION TITLE
Does code-server supportes multiple work?

### DISCUSSION THREAD

**Discussion by stevalkr** [coder/code-server#3293] [Q&A]
We are working as a driverless racing team and we need to connect and edit the codes remotely on the workstation. By using code-server, every time the page is refreshed, it would be redirected to someone else's.   
I've tried starting multi sessions in different ports but it didn't help. Docker may not be possiable to use because full resources of computer is needed for the programs.  
How to deal with that?

**Comment by code-asher** [comment]
My guess would be that it happens because they are all sharing the same data directory (which includes settings, the configuration file that contains the last opened workspace, etc).

So one solution would be to start on different ports and with different data directories for each user.

```
code-server --port <port> --user-data-dir <dir>
```

**Reply by stevalkr** [reply]
I‘ve tried this but forgot to tell. Changing this data folder only makes appearance different in each port. It still opens the newest workfolder and the opened files automatically.

**Reply by code-asher** [reply]
I might be misunderstanding but yes, that's by design. Each user will need their own port and data directory.
